### ISSUE TITLE
Material ball scene seems to introduce spreading NaNs in SH mode with both ReLAX and ReBLUR

### ISSUE THREAD

**Issue by Nielsbishere** [NVIDIAGameWorks/NRDSample#15]
I've been able to see NaNs spread using NRD_MODE == SH in both ReLAX and ReBLUR if you fly around enough. 

In the USE_SANITATION flag it specifically says that the NRDSample doesn't have any NaNs/Infs, which is clearly wrong, because after I enable it, it works again. I'm unsure if this means USE_SANITATION needs to be forced on for NRDSample or if the NaNs/Infs themselves have to be fixed in the NRDSample itself.

**Comment by dzhdanNV** [comment]
Unfortunately, I can't reproduce the problem. Tried:
- D3D12/VK
- debug/release
- RELAX/REBLUR
- lots of flying around
- various tracing options
- USE_SANITIZATION = off, SH = on, resolve = on

**Comment by Nielsbishere** [comment]
Hi, sorry for the late reply. I have tried to repro it multiple times but somehow I don't seem to be able to.
Luckily I've taken screencaps when it happened on the latest version at the time.
Here's more information about the environment:
Environment: scene: Shader balls, Backend: D3D12, Release: v4.4.1, GPU: RTX 4090, Driver: 551.23, OS: Windows 10 22H2 (19045.3996), flags: #define NRD_MODE SH, #define USE_SANITIZATION 0, resolution: 1440p. Running through "3-Run NRD sample.bat" with options 1, 2, 1, 1, 3.

Here's what I got back then:
![image (48)](https://github.com/NVIDIAGameWorks/NRDSample/assets/28758736/b7282020-5de9-4fce-a2e4-acb9c1da8a45)
![image (49)](https://github.com/NVIDIAGameWorks/NRDSample/assets/28758736/42f5e13d-b189-4c5e-849c-206fb67eb242)

The only thing I was able to consistently reproduce seems to be a different artifact that has nothing to do with NaNs. My best guess is it is some sort of precision issue that causes self intersection (maybe viewZ isn't precise enough or something?).
![ezgif-3-2934ea169c](https://github.com/NVIDIAGameWorks/NRDSample/assets/28758736/4fd6387f-0064-4244-a93e-a1ee2889a757)
Though this is not a NaN, I'll re-open if I can ever find a solid repro.
